src/elf/dyn.rs

```diff
@@ -624,17 +609,52 @@ macro_rules! elf_dyn_std_impl {
                     _ => (),
                 }
             }
-            pub fn new(dynamic: &[Dyn], bias: usize) -> DynamicInfo {
+            pub fn new(dynamic: &[Dyn], phdrs: &[$phdr]) -> DynamicInfo {
                 let mut info = DynamicInfo::default();
                 for dyn in dynamic {
-                    info.update(bias, &dyn);
+                    info.update(phdrs, &dyn);
                 }
                 info
             }
-        } // end if_std
+        }
+
+        if_alloc! {
```

Comment:
Just curious: how come this got moved to alloc?

Reply:
It was always in alloc. I just moved it into another macro and kept it in alloc without thinking about whether it should be. I can move it out as part of this PR if you want, but it's probably better as another PR.